### ISSUE TITLE
bug: Treesitter highlight in Python files

### ISSUE THREAD

**Issue by dpetka2001** [LazyVim/LazyVim#1602]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.10.0-dev-1287+ga6cccd728

### Operating system/version

Linux Mint 21.1

### Describe the bug

When opening a Python file I get the following error
```
Error in decoration provider treesitter/highlighter.line:
Error executing lua: /usr/share/nvim/runtime/lua/vim/treesitter/query.lua:248: Query error at 30:2. Invalid node type "_":
"_" @constant.builtin ; match wildcard
 ^

stack traceback:
	[C]: in function '_ts_parse_query'
	/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:248: in function 'fn'
	/usr/share/nvim/runtime/lua/vim/func/_memoize.lua:54: in function 'fn'
	/usr/share/nvim/runtime/lua/vim/func/_memoize.lua:54: in function 'get'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:53: in function 'new'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:206: in function 'get_query'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:231: in function 'fn'
	/usr/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:483: in function 'for_each_tree'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:217: in function 'on_line_impl'
	/usr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:298: in function </usr/shar
```
The problem is happening only when providing Neovim with an argument, like so `nvim test.py`. When you invoke Neovim without arguments with just `nvim` and then either use `find file` or `restore session` to open the file, no error occurs.

### Steps To Reproduce

1. Invoke nvim with `nvim test.py` and you get the error
2. Invoke nvim without arguments with just `nvim` and then choose either `find file` or `restore session` from alpha dashboard to choose the Python file and no errors occur

### Expected Behavior

No errors.

As a workaround I changed `nvim-treesitter`'s event to `event = { "BufReadPost", "BufNewFile" }`, like it used to be, and I get no error when opening a file either with or without arguments.

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/LazyVim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by folke** [comment]
I actually made a bunch of fixes locally for lazy and LazyVim related to LAzyFile, but I would still like to reproduce your issue here.

With stock LazyVim and the python extra I can't seem to reproduce your issue.

Does it trigger with an empty python file too?

**Comment by amaanq** [comment]
Probably one of two things:

You need to have python in nvim-treesitter's ensure_installed table (which it is by default but perhaps you're overriding it and it's not there anymore)

You haven't run :TSUpdate after updating nvim-treesitter

**Comment by folke** [comment]
It's definitely possible that there is a bug in the lazy/LazyVim code right now.

I just changed the implementation of lazy's event handler to be more robust. Also fixes some bugs.

So I probably fixed @dpetka2001's issue, but want to be sure before I commit my changes :)

**Comment by dpetka2001** [comment]
@folke I was able to reproduce with a fresh Lazyvim starter template as it currently is, apart from my config to eliminate an error in my configuration.
```lua
1. mkdir ~/.config/newstarter && cd ~/.config/newstarter
2. git clone https://github.com/LazyVim/starter .
3. `NVIM_APPNAME=newstarter nvim` and wait for installation to complete
4. Restart Neovim with `NVIM_APPNAME=newstarter nvim test.py`
5. Press `i` to enter `insert` mode and start typing `print("Hello, World!")` and notice the errors.
```

@amaanq My parsers are up to date and `TSUpdate` is handled by LazyVim already when updating the plugin.

**Comment by folke** [comment]
hmm, still can't reproduce with those steps. very odd

**Comment by dpetka2001** [comment]
Is there anything else I could try to provide you additional info?

**Comment by folke** [comment]
I just pushed changes for lazy.nvim and LazyVim.

I won't make a new release yet. 

Can you test if these changes also fix your issue?

**Comment by folke** [comment]
To use the dev versions: https://www.lazyvim.org/configuration/lazy.nvim

**Comment by dpetka2001** [comment]
Nope, still getting the error with the `newstarter` that I installed in my previous post.

**Comment by folke** [comment]
newstarter won't use the dev version

**Comment by folke** [comment]
Add this to newstarter:

```lua
return {
  { "folke/lazy.nvim", version = false },
  { "LazyVim/LazyVim", version = false },
}
```

And then open newstarter and upgrade

**Comment by dpetka2001** [comment]
Yes, i know, i added a file `core.lua` with contents
```lua
return {
  { "folke/lazy.nvim", version = false },
  { "LazyVim/LazyVim", version = false },
}
```
and then it updated to latest commits for both of them. Still, no change.

**Comment by folke** [comment]
really wierd. Can you test on latest stable NEovim or latest nightly? Probably something off with your version

**Comment by folke** [comment]
I just pushed a new release for both lazy and LazyVim.

**Comment by dpetka2001** [comment]
I'm on latest HEAD. commit `5db076c`

**Comment by folke** [comment]
do you have any custom treesitter queries?

**Comment by dpetka2001** [comment]
No custom queries. With the `newstarter` I was just using default Lazyvim configuration.

**Comment by folke** [comment]
works for me on latest nightly and stable. I use bob for switching Neovim versions.

**Comment by folke** [comment]
How do you install Neovim?

You seem to be using a globally installed runtime: /usr/share/nvim/runtime/lua/vim/treesitter/query.lua

Is that correct?

**Comment by dpetka2001** [comment]
I build it from source. I don't where that path is coming from. I'm guessing from the default installation?

**Comment by dpetka2001** [comment]
@folke I uninstalled Neovim completely and deleted the Neovim repo as well. Then re-cloned the repo and followed the instructions for building Neovim from source again and I can confirm it's working as expected with exact the same configuration as before. Thank you for pointing out about the Neovim version. I wouldn't have thought of that. I also don't know what went wrong with building it from source, like I used to do until now. I made sure to delete the whole repo and re-cloned it, but I don't believe this should be happening. I will also look into `bob` like you suggested. Tnx a lot again for your help and patience.
